examples/client/delete_empty_project_versions.py

```diff
@@ -0,0 +1,52 @@
+from blackduck import Client
+from blackduck.Utils import to_datetime
+
+from datetime import datetime, timedelta
+import argparse
+import logging
+import requests
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(asctime)s] {%(module)s:%(lineno)d} %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser("delete_empty_project_versions")
+parser.add_argument("--base-url", required=True, help="blackduck hub server url e.g. https://your.app.blackduck.com")
+parser.add_argument("--token-file", dest='token_file', required=True, help="path to file containing blackduck hub token")
+parser.add_argument("--no-verify", dest='verify', action='store_false', help="disable TLS certificate verification")
+parser.add_argument("--days", dest='days', default=30, type=int, help="projects/versions older than <days>")
+parser.add_argument("--delete", dest='delete', action='store_true', help="without this flag script will log/print only")
+args = parser.parse_args()
+
+with open(args.token_file, 'r') as tf:
+    access_token = tf.readline().strip()
+
+bd = Client(
+    base_url=args.base_url,
+    token=access_token,
+    verify=args.verify
+)
+
+max_age = datetime.now() - timedelta(days=args.days)
+
+for project in bd.get_resource('projects'):
+    # skip projects younger than max age
+    if to_datetime(project.get('createdAt')) > max_age: continue
```

Comment:
I'm getting an exception here: TypeError: can't compare offset-naive and offset-aware datetimes

Reply:
A possible solution could be to create something like this:

```
def to_local_datetime(date):
    return to_datetime(date).astimezone(tz=tzlocal())   

def get_local_datetime():
    return datetime.now(tz=tzlocal())
```

Normally I'd avoid creating wrappers for stl functions but in this case I don't think it is obvious how to get a timezone aware datetime given  `datetime.now()` and even `datetime.utcnow()` are timezone naive. Thoughts?

Reply:
In [sage_version_activity_to_csv.py](https://github.com/blackducksoftware/sage/blob/master/sage_version_activity_to_csv.py) I used: from dateutil.parser import isoparse
Would that work?

Reply:
I'll take a look at that tonight